packages/analytics-core/src/utils/uuid.ts

```diff
@@ -1,32 +1,21 @@
 /**
- * Source: [jed's gist]{@link https://gist.github.com/982883}.
+ * Source: [jed's gist's comment]{@link https://gist.github.com/jed/982883?permalink_comment_id=3223002#gistcomment-3223002}.
  * Returns a random v4 UUID of the form xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx,
  * where each x is replaced with a random hexadecimal digit from 0 to f, and
  * y is replaced with a random hexadecimal digit from 8 to b.
  * Used to generate UUIDs for deviceIds.
  * @private
  */
-export const UUID = function (a?: any): string {
-  return a // if the placeholder was passed, return
-    ? // a random number from 0 to 15
-      (
-        a ^ // unless b is 8,
-        ((Math.random() * // in which case
-          16) >> // a random number from
-          (a / 4))
-      ) // 8 to 11
-        .toString(16) // in hexadecimal
-    : // or otherwise a concatenated string:
-      (
-        String(1e7) + // 10000000 +
-        String(-1e3) + // -1000 +
-        String(-4e3) + // -4000 +
-        String(-8e3) + // -80000000 +
-        String(-1e11)
-      ) // -100000000000,
-        .replace(
-          // replacing
-          /[018]/g, // zeroes, ones, and eights with
-          UUID, // random hex digits
-        );
+import * as crypto from 'crypto';
```

Comment:
is this package available on the browser?

Reply:
should we use `window.crypto`?

Reply:
also found out that there's `window.crypto.randomUUID()` that returns a random UUID

Reply:
It's available on browser (https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues). 

This import is for nodejs environment as `yarn test` returns `Cannot find name 'crypto'` error without importing it. 

`yarn test` runs successfully locally.

Reply:
i think this would throw an error on prod since crypto package does not exist. i think you mean `window.crypto`

Reply:
one if the pr checks also caught the error

Reply:
I looked at this post https://stackoverflow.com/questions/70033580/what-is-the-difference-between-import-crypto-from-crypto-vs-just-using-crypto:
- For browser environment, `crypto` is predefined globals. That's why it can be used directly in [JS SDK](https://github.com/amplitude/Amplitude-JavaScript/pull/594)
- For node environment, `crypto` is not predefined

Reply:
synced offline and determined that the build throws an error 
![image](https://github.com/amplitude/Amplitude-TypeScript/assets/20634289/6d8badcd-2756-4a83-ba33-94cb674f94f5)